sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorServiceImpl.java

```diff
@@ -37,15 +42,20 @@ public long createSponsor(@NotBlank String name, @NotBlank String email)
   }
 
 
-  @Override
-  public void sendEmailNotification(Sponsor sponsor)
+  private void sendEmailNotification(Sponsor sponsor)
   {
-    try {
-      emailClient.sendNotification(sponsor.toEmailMessage());
+    EmailMessage emailMessage = sponsor.toEmailMessage();
+    if (isSyncNotification) {
+      try {
+        emailClientSync.sendNotification(emailMessage);
+      }
+      catch (Exception e) {
+        //todo it is workaround, what behavior is expected when the other service is not available?
+        e.printStackTrace();
+      }
     }
-    catch (Exception e) {
-      //todo it is workaround, what behavior is expected when the other service is not available?
-      e.printStackTrace();
+    else {
+      emailClientAsync.sendNotification(emailMessage);

```

Comment:
I think this logic should be placed in NotificationSenderService or EmailClient. And it's responsibility of this Service/Client to decide.

Reply:
Good suggestion, thanks. Done.